src/proc/formats-converter.cpp

```diff
@@ -0,0 +1,396 @@
+            target_profiles.erase( std::remove_if( begin( target_profiles ), end( target_profiles ), matching_profiles_predicate ) );
+        }
+
+        // Retrieve source profile from cached map and generate the relevant processing block.
+        std::unordered_set<std::shared_ptr<stream_profile_interface>> current_resolved_reqs;
+        auto best_pb = best_match_pbf->generate();
+        for( const auto & target_profile : best_match_target_profiles )
+        {
+            auto & mapped_source_profiles = _target_to_source_profiles_map[to_profile( target_profile.get() )];
+
+            for( const auto & source_profile : mapped_source_profiles )
+            {
+                if( best_match_pbf->has_source( source_profile ) )
+                {
+                    current_resolved_reqs.insert( source_profile );
+
+                    // Caching processing blocks to set their callback during sensor::start
+                    _source_profile_to_converters[source_profile].insert( best_pb );
+                }
+            }
+        }
+        const stream_profiles & print_current_resolved_reqs = { current_resolved_reqs.begin(), current_resolved_reqs.end() };
+        LOG_INFO( "Request: " << best_match_target_profiles << "\nResolved to: " << print_current_resolved_reqs );
+    }
+}
+
+void formats_converter::update_source_data( const stream_profiles & target_profiles )
+{
+    for( auto & target_profile : target_profiles )
+    {
+        // TODO - original code only updates source[0], why?
+        for( auto & source_profile : _target_to_source_profiles_map[to_profile( target_profile.get() )] )
+        {
+            source_profile->set_unique_id( target_profile->get_unique_id() );
+            source_profile->set_stream_index( target_profile->get_stream_index() );
+            source_profile->set_unique_id( target_profile->get_unique_id() );
+            source_profile->set_stream_type( target_profile->get_stream_type() );
+            auto source_video_profile = As< video_stream_profile, stream_profile_interface >( source_profile );
+            const auto target_video_profile = As< video_stream_profile, stream_profile_interface >( target_profile );
+            if( source_video_profile )
+            {
+                source_video_profile->set_intrinsics( [target_video_profile]()
+                {
+                    if( target_video_profile )
+                        return target_video_profile->get_intrinsics();
+                    else
+                        return rs2_intrinsics{};
+                } );
+
+                // Hack for L515 confidence.
+                // Requesting source resolution from the camera, getting frame size of target (*2 y axis resolution)
+                source_video_profile->set_dims( target_video_profile->get_width(), target_video_profile->get_height() );
+            }
+        }
+    }
+}
+
+void formats_converter::cache_target_profiles( const stream_profiles & target_profiles )
```

Comment:
Extracted from `prepare_to_convert`. More readable and I think that after the refactoring I want to do it won't be necessary.